tla-io/src/main/scala/at/forsyte/apalache/io/quint/Quint.scala

```diff
@@ -458,11 +470,33 @@ class Quint(moduleData: QuintOutput) {
       applicationBuilder(quintArgs)
     }
 
+    // Convert Quint's nondet variable binding
+    //
+    //   val nondet name = oneOf(domain); scope
+    //   ~~>
+    //   LET wrap = \E name \in domain: scope IN wrap
+    //
+    // We must wrap the binding, because this is at the declaration level of the quint
+    // IR, but must be at the expression level in the Apalache IR.
```

Comment:
@Kukovec probably remembers it better, but I think that wrapping an existential in `LET` may prevent the preprocessor from applying Skolemization. I do not see why we could not simply produce an expression. `nondet` declarations cannot appear as top-level declarations in Quint, they only appear inside other operator declarations. Nested `LET-IN` expressions are just expressions.

Reply:
Yep! Thanks for catching that. I'm not sure why I'd thought it needed wrapping. Fixed.